docs/k8s/deployment-guide.mdx

```diff
@@ -0,0 +1,165 @@
+- helm - 3.0.0 or later.
+
+## Installation
+
+It is recommended to use helm to install the operator. Alternatively, the container is available on docker hub at `ngrok/ingress-controller` and can be run directly with hand crafted manifests.
+
+<Tabs groupId="k8s" queryString="k8s-basic">
+	<TabItem value="ingress" label="Ingress Controller" default>
+		<K8sIngressInstallExample />
+	</TabItem>
+	<TabItem value="gatewayAPI" label="Gateway API">
+		<K8sGatewayAPIInstallExample />
+	</TabItem>
+</Tabs>
+
+For a more robust _infrastructure as code_ way of passing your credentials, see the [credentials](#credentials-setup) page.
+
+### Alternate options
+
+<Tabs groupId="k8s" queryString="k8s-basic">
+	<TabItem value="ingress" label="Ingress Controller" default>
+For a quick install, you can also use the combined manifests directly from the repo and begin changing resources:
+
+```bash
+kubectl apply -n ngrok-ingress-controller -f https://raw.githubusercontent.com/ngrok/kubernetes-ingress-controller/main/manifest-bundle.yaml
+```
+
+  </TabItem>
+  <TabItem value="gatewayAPI" label="Gateway API">
+::::note
+
+The above solution is the recommended way to set up the Gateway API
+
+::::
+
+  </TabItem>
+</Tabs>
+
+## Credentials
+
+In order to use the ngrok Kubernetes Operator, you will need an ngrok account. Once you have an account, you will need to log into the [ngrok dashboard](https://dashboard.ngrok.com) to gather the necessary credentials.
+
+You will need two things from the dashboard:
+
+- Your auth token, which can be found [here](https://dashboard.ngrok.com/auth/your-authtoken)
+- An API key, which can be found [here](https://dashboard.ngrok.com/api)
+
+These credentials will be created as a Kubernetes secret, which the controller will have access to. The auth token is used to create tunnels, and the API key is used to manage edges and other resources via the ngrok API.
+
+### Setup {#credentials-setup}
+
+While the quickstart guide shows you can pass the credential values directly via helm values, we do not recommend this for production scenarios. Instead, we recommend creating a Kubernetes secret and passing the secret name to the helm chart. This allows for easier infrastructure as code in a more secure manner.
+
+### Creating the Secret
+
+To create the secret, follow these steps:
+
+- Make sure the secret is in the same namespace as the ingress controller
+- Use a well-formed name that can be passed to the helm chart
+- Add two keys to the secret: `API_KEY` and `AUTHTOKEN`
+
+Here is an example secret manifest:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ngrok-ingress-controller-credentials
+  namespace: ngrok-ingress-controller
+data:
+  API_KEY: "YOUR-API-KEY-BASE64"
+  AUTHTOKEN: "YOUR-AUTHTOKEN-BASE64"
+```
+
```

Comment:
I think this belongs in the usage guide rather than the deployment guide since it is more about how to use a capability via the manifest rather than how to set up the operator.